src/main/java/de/tum/in/www1/artemis/web/rest/settings/ide/IdeSettingsResource.java

```diff
@@ -0,0 +1,145 @@
+     */
+    @PutMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<IdeMappingDTO> setIde(@RequestParam ProgrammingLanguage programmingLanguage, @RequestBody IdeDTO ide) {
+        User user = userRepository.getUser();
+        log.debug("REST request to set IDE of user {}", user.getLogin());
+        if (ide == null || ide.deepLink() == null || ide.name() == null) {
+            log.error("Invalid request body for IDE setting of user {}", user.getLogin());
+            return ResponseEntity.badRequest().build();
+        }
+        if (!ide.deepLink().contains("://") || !ide.deepLink().contains("{cloneUrl}")) {
+            log.error("Invalid deep link for IDE {} of user {}", ide.deepLink(), user.getLogin());
+            return ResponseEntity.badRequest().build();
+        }
+
+        var oldIdeMapping = userIdeMappingRepository.findById(new UserIdeMappingId(user.getId(), programmingLanguage)).orElse(null);
+
+        // find or create the ide
+        var savedIde = ideRepository.findByDeepLink(ide.deepLink()).orElse(null);
+        if (savedIde == null) {
+            savedIde = ideRepository.save(new Ide(ide.name(), ide.deepLink()));
+        }
+
+        // Create or update user ide mapping
+        UserIdeMapping ideMapping = new UserIdeMapping(user, programmingLanguage, savedIde);
+        ideMapping = userIdeMappingRepository.save(ideMapping);
+
+        orphanRemoval(oldIdeMapping != null ? oldIdeMapping.getIde() : null);
+
+        var ideRecord = new IdeMappingDTO(ideMapping.getProgrammingLanguage(), ideMapping.getIde());
+        log.debug("Successfully set IDE of user {}", user.getLogin());
+
+        return ResponseEntity.ok(ideRecord);
+    }
+
+    /**
+     * DELETE ide-settings: delete a programming language from a users ide preferences
+     *
+     * @param programmingLanguage the programming language preference to delete
+     * @return the ResponseEntity with status 200 (OK), or with status 400 (Bad Request)
+     */
+    @DeleteMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Void> deleteProgrammingLanguageOfUser(@RequestParam ProgrammingLanguage programmingLanguage) {
+        User user = userRepository.getUser();
+        log.debug("REST request to delete IDE of user {}", user.getLogin());
+
+        var ideMapping = userIdeMappingRepository.findById(new UserIdeMappingId(user.getId(), programmingLanguage)).orElse(null);
+        if (ideMapping == null) {
+            log.error("No IDE found for user {} with programming language {}", user.getLogin(), programmingLanguage);
+            return ResponseEntity.badRequest().build();
+        }
+
+        userIdeMappingRepository.deleteById(new UserIdeMappingId(user.getId(), programmingLanguage));
+
+        orphanRemoval(ideMapping.getIde());
+
+        log.debug("Successfully deleted IDE of user {}", user.getLogin());
+        return ResponseEntity.ok().build();
+    }
+
+    // if ide is parentless, delete it
```

Comment:
I'd prefer a proper JavaDoc instead of a normal comment when explaining methods.